hibernate-core/src/test/java/org/hibernate/test/annotations/inheritance/joined/JoinedSubclassAndSecondaryTable.java

```diff
@@ -6,100 +6,121 @@
-
-	@Test
-	public void testSecondaryTableAndJoined() {
-		Session s = openSession();
-		Transaction tx = s.beginTransaction();
-		SwimmingPool sp = new SwimmingPool();
-		s.persist( sp );
-		s.flush();
-		s.clear();
-
-		long rowCount = getTableRowCount( s, "POOL_ADDRESS" );
-		assertEquals(
-				"The address table is marked as optional. For null values no database row should be created",
-				0,
-				rowCount
-		);
-
-		SwimmingPool sp2 = (SwimmingPool) s.get( SwimmingPool.class, sp.getId() );
-		assertNull( sp.getAddress() );
-
-		PoolAddress address = new PoolAddress();
-		address.setAddress( "Park Avenue" );
-		sp2.setAddress( address );
-
-		s.flush();
-		s.clear();
-
-		sp2 = (SwimmingPool) s.get( SwimmingPool.class, sp.getId() );
-		rowCount = getTableRowCount( s, "POOL_ADDRESS" );
-		assertEquals(
-				"Now we should have a row in the pool address table ",
-				1,
-				rowCount
+@DomainModel(
+		annotatedClasses = {
+				Pool.class,
+				SwimmingPool.class
+		}
+)
+@SessionFactory
+public class JoinedSubclassAndSecondaryTable {
+
+	@AfterEach
+	public void tearDown(SessionFactoryScope scope) {
+		scope.inTransaction(
+				sesison ->
+						sesison.createQuery( "from Pool" ).list().forEach(
+								pool -> sesison.delete( pool )
+						)
 		);
-		assertNotNull( sp2.getAddress() );
-		assertEquals( sp2.getAddress().getAddress(), "Park Avenue" );
-
-		tx.rollback();
-		s.close();
 	}
 
 	@Test
-	public void testSecondaryTableAndJoinedInverse() throws Exception {
-		Session s = openSession();
-		Transaction tx = s.beginTransaction();
-		SwimmingPool sp = new SwimmingPool();
-		s.persist( sp );
-		s.flush();
-		s.clear();
-
-		long rowCount = getTableRowCount( s, "POOL_ADDRESS_2" );
-		assertEquals(
-				"The address table is marked as optional. For null values no database row should be created",
-				0,
-				rowCount
+	public void testSecondaryTableAndJoined(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					SwimmingPool sp = new SwimmingPool();
+					session.persist( sp );
+					session.flush();
+					session.clear();
+
+					long rowCount = getTableRowCount( session, "POOL_ADDRESS" );
+					assertEquals(
+
+							0,
+							rowCount,
+							"The address table is marked as optional. For null values no database row should be created"
+					);
+
+					SwimmingPool sp2 = session.get( SwimmingPool.class, sp.getId() );
+					assertNull( sp.getAddress() );
+
+					PoolAddress address = new PoolAddress();
+					address.setAddress( "Park Avenue" );
+					sp2.setAddress( address );
+
+					session.flush();
+					session.clear();
+
+					sp2 = session.get( SwimmingPool.class, sp.getId() );
+					rowCount = getTableRowCount( session, "POOL_ADDRESS" );
+					assertEquals(
+
+							1,
+							rowCount,
+							"Now we should have a row in the pool address table "
+
+					);
+					assertNotNull( sp2.getAddress() );
+					assertEquals( sp2.getAddress().getAddress(), "Park Avenue" );
```

Comment:
seems `Park Avenue` is the expected value and should be put as the first parameter. Not a big deal for sure.

Reply:
Fixed, thanks Nathan